### DISCUSSION TITLE
v0.12.0-beta.1

### DISCUSSION THREAD

**Discussion by alexjustesen** [alexjustesen/speedtest-tracker#730] [Releases]
## What's Changed
* [Feature] PHP 8.2 and performance improvements by @alexjustesen in https://github.com/alexjustesen/speedtest-tracker/pull/676
* Release v0.12.0-beta.1 by @alexjustesen in https://github.com/alexjustesen/speedtest-tracker/pull/729

**Full Changelog**: https://github.com/alexjustesen/speedtest-tracker/compare/v0.11.22...v0.12.0-beta.1

<hr /><em>This discussion was created from the release <a href='https://github.com/alexjustesen/speedtest-tracker/releases/tag/v0.12.0-beta.1'>v0.12.0-beta.1</a>.</em>

**Comment by ShlomiD83** [comment]
@alexjustesen I think loading times on the RPi 4 are wicked fast now, I'm talking a few seconds.

**Reply by alexjustesen** [reply]
F YEAH! Glad to hear that, should be instant on most other devices too

**Reply by ShlomiD83** [reply]
Thanks!

**Reply by corasaniti** [reply]
@alexjustesen 
We can we trust the beta version? Wouldn't it be better to have a beta tag for the Docker Container and keep the stable version on the latest tag? Thanks

**Reply by jiriteach** [reply]
Seems to be working ok so far. Just needed 2 reboots due to APP_KEY.

**Reply by alexjustesen** [reply]
> @alexjustesen 
> We can we trust the beta version? Wouldn't it be better to have a beta tag for the Docker Container and keep the stable version on the latest tag? Thanks

I agree it shouldn't be tagged as `:latest` I need to get the build process updated to account for other builds.

**Comment by jiriteach** [comment]
Performance has improved which is really good. Great work on this. 

It feels faster. I’ve been looking into other bits that could cause slow loads and I see there are a few things mentioned with Filament and Laravel. 

Worth looking into as well?
https://github.com/filamentphp/filament/discussions/5654

**Reply by alexjustesen** [reply]
Queries could absolutely be one of the things and requests to the server. Current build have Filament `v2`, I'm working on moving to `v3` over the next couple of days so be interesting to see if we see performance improvements there too.

**Reply by jiriteach** [reply]
Awesome - look forward to testing :)

**Comment by corasaniti** [comment]
@Jiriteach @alexjustesen 
I agree but the latest docker tag should be for the stable version and not for the beta editions...

**Comment by alexdelprete** [comment]
> @Jiriteach @alexjustesen I agree but the latest docker tag should be for the stable version and not for the beta editions...

Never use latest tag if you're concerned about stability etc. that's basics 101 for docker images. Always specify version tags, and use a tool to manage notifications of new versions (I use What's Up Docker).

That would prevent things like this happening. But the cost for more reliability is obviously more maintenance tasks.

**Comment by jiriteach** [comment]
To clarify - I agree that a beta tag should be used instead of pushing to latest. Latest should remain for the latest stable version.

With some of the bigger changes you are working on @alexjustesen - a beta tag would definitely make sense. Those that want to can use this beta tag as need for testing.

If there was one - I personally would setup another instance against beta to test and keep my other versions on the latest stable.

**Comment by alexdelprete** [comment]
> a beta tag would definitely make sense. Those that want to can use this beta tag as need for testing.

The beta tag exists, check release page on the repo: https://github.com/alexjustesen/speedtest-tracker/releases

```
version: '3.3'
services:
  speedtest-tracker:
    image: ghcr.io/alexjustesen/speedtest-tracker:v0.12.0-beta.1
    container_name: speedtest-tracker
    restart: unless-stopped

```

**Reply by jiriteach** [reply]
It does for that specific release. There are multiple ways of handling this and each have pro's and con's. Yes one way is not to the use the latest tag. On other repo's - pulling from latest only pulls stable versions and any pre-release or beta needs to explicitly use that versions release tag.

**Reply by alexdelprete** [reply]
No, "latest" is always to be considered unreliable because it is subjective: for a dev it might mean bleeding-edge, for another means stable, that's why the best practice is to use fixed versions of the tags.

The problem you saw for ST for "latest" is exactly why the best practice is always recommended:

![image](https://github.com/alexjustesen/speedtest-tracker/assets/7027842/f35fda95-8618-4161-a26b-edb4b4dc1081)

**Reply by jiriteach** [reply]
Sure - at the end of the day its ultimately the users (or those pulling a version) that define this. LIke the best practice advise says - fixate the version. By default - even if you tell people to pull a specific version - some might just use latest and its something for them to handle and deal with so perhaps its just a simple tweak to documentation around the emphasising the use of the specific version. 
I'm sure we'll still see users pulling the latest tag and seeing issues - some will follow and and many wont.

**Comment by corasaniti** [comment]
@alexjustesen @alexdelprete @Jiriteach 

In my opinion two tags should be created
```
stable-edition:latest
stable-edition:0.11.22
```

Also another tag for the unstable development version
`unstable-edition:0.12-beta1`

The user by default pulls the latest tag and therefore a stable version must be guaranteed. If you want to try a version that is still in development, you will need to pull the 0.12-beta1 tag
Thank you for the excellent work done...
PC

**Comment by alexdelprete** [comment]
> Also another tag for the unstable development version

the tag is the part to the right of the : on the left you have the repo name: `reponame:tag`.

The tag already contains beta indicating an unstable version, and labeled also as pre-release in the release page.

![image](https://github.com/alexjustesen/speedtest-tracker/assets/7027842/1cb245c9-284d-48c6-ae74-3f41dc6cd228)

It's up to the maintainer what to associate the "latest" tag to, there's no strict rule. I'm sure @alexjustesen will avoid tagging betas as latest in the future.

**Comment by alexjustesen** [comment]
Closing convo, follow https://github.com/alexjustesen/speedtest-tracker/pull/739 for changes to the build tags.
